velox/expression/SwitchExpr.cpp

```diff
@@ -90,6 +90,18 @@ void SwitchExpr::evalSpecialForm(
     if (!remainingRows.get()->hasSelections()) {
       break;
     }
+
+    if (condition) {
```

Comment:
you can just use the following API to ensure re-use. This will avoid the copy with non-flat vectors as well so no need to handle different cases :
```
static void prepareForReuse(VectorPtr& vector, vector_size_t size);
```